### PR TITLE
AI Chat: basic UI tests

### PR DESCRIPTION
**Pull request by bencodeorg**
Adds initial UI tests that confirm a (levelbuilder) user can:
- Roundtrip a chat completion request [edit: skipped in Drone because of issues accessing SageMaker, [Slack discussion here](https://codedotorg.slack.com/archives/C03CK49G9/p1725475362107969)]
- Update model customizations (eg, system prompt) and reload them on page refresh
- Fill out a model card and see the "presentation view", and reload that model card on page refresh

I didn't add any eyes tests (I think we can wait a little to add those as we finalize the UI), and I think there are probably a couple more detailed scenarios we could add (content moderation of model customizations and chat requests?), but I wanted to at least get an end to end test up that would alert us if the "base case" (ie, being able to interact with the model) wasn't working.

## Links

- jira: https://codedotorg.atlassian.net/browse/LABS-818

## Follow-up work

I think (?) we are planning on setting up a test instance for our model(s), yeah? We should ideally be pointing our tests at that instance -- currently, tests run against our production instance.

### CODE REVIEW

**Comment by bencodeorg** [comment]
@levadadenys I added the option for an ID prop in a few components in the component library here, LMK if you have any objections :)

File: dashboard/test/ui/features/star_labs/aichat/chat.feature [hunk 1]
```diff
@@ -0,0 +1,63 @@
+@no_mobile
+Feature: AI Chat
+
+  "AI Chat" is our lab that introduces students to generative AI
```
Comment (fisher-alice) [inline dashboard/test/ui/features/star_labs/aichat/chat.feature:4]: Do we not need to comment this section?

File: dashboard/test/ui/features/star_labs/aichat/chat.feature [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  # As of 9/4/24, cannot access SageMaker in Drone.
+  # More discussion in this Slack thread: https://codedotorg.slack.com/archives/C03CK49G9/p1725475362107969
+  @no_circle
+  Scenario: Making chat request gets response
+    When I press keys "Hello" for element "#uitest-chat-textarea"
+    And I wait until element "#uitest-chat-submit" is enabled
+    And I click selector "#uitest-chat-submit"
+    And I wait until element "[aria-label='AI bot']" is visible
+    Then element "[aria-label='AI bot']" has css property "background-color" equal to "rgb(224, 248, 249)"
```
Comment (fisher-alice) [inline dashboard/test/ui/features/star_labs/aichat/chat.feature:22]: Nice! Was thinking about the error case below and cool that you check the background color of assistant message to confirm it's successful. 

Screenshot 2024-09-06 at 10 20 43 AM

File: dashboard/test/ui/features/star_labs/aichat/chat.feature [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  Scenario: Editing system prompt produces success notification and saves
+    When I press keys "You are a silly chatbot" for element "#system-prompt"
```
Comment (fisher-alice) [inline dashboard/test/ui/features/star_labs/aichat/chat.feature:25]: Maybe add check that element `"#system-prompt"` is blank  for readability? Wasn't sure what the initial text was until watching video.

File: dashboard/test/ui/features/star_labs/aichat/chat.feature [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    And I wait until element "#uitest-update-customizations" is enabled
+    And I click selector "#uitest-update-customizations"
+    Then I wait until element ".uitest-aichat-chat-alert" contains text "System prompt has been updated"
+
+    Given I reload the page
```
Comment (fisher-alice) [inline dashboard/test/ui/features/star_labs/aichat/chat.feature:30]: Just a question for my understanding, is there a difference between 'Given' vs 'And' or 'When' here?

**Review by fisher-alice — APPROVED** [review]
🎉

File: dashboard/test/ui/features/star_labs/aichat/chat.feature (hunk 4 above)
Comment (bencodeorg) [reply dashboard/test/ui/features/star_labs/aichat/chat.feature:30]: hah good q, I was trying to follow the guidance from the Cucumber docs about when to use given/when/then (I always get them mixed up):

https://cucumber.io/docs/gherkin/reference/#given

It's a little subjective, and doesn't have any code impact (just for readability).

File: dashboard/test/ui/features/star_labs/aichat/chat.feature (hunk 1 above)
Comment (bencodeorg) [reply dashboard/test/ui/features/star_labs/aichat/chat.feature:4]: We do not, another thing I learned from the cucumber docs is that you can include descriptions like these and they'll show up at the top of your test output:

image

https://cucumber.io/docs/gherkin/reference/#feature

File: dashboard/test/ui/features/star_labs/aichat/chat.feature (hunk 3 above)
Comment (bencodeorg) [reply dashboard/test/ui/features/star_labs/aichat/chat.feature:25]: Sorry not sure I follow, is the initial text something that we care about?

Comment (fisher-alice) [reply dashboard/test/ui/features/star_labs/aichat/chat.feature:25]: When I read the test steps, I was wondering about the case of the default system prompt already set by the levelbuilder to be 'You are a silly chatbot'. So wanted to confirm that the system prompt changed and was saved. Definitely a nit and no need to change! It just prompted me to go look at the video and see the system prompt default is set to `""` initially.

**Comment by fisher-alice** [comment]
Just another thank you for adding these tests! Helped to catch a really sneaky bug https://github.com/code-dot-org/code-dot-org/pull/60978.

**Pull request merged**